### PR TITLE
Typed testing docs: imports and scalatest dependency

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java [hunk 1]
```diff
@@ -9,11 +9,17 @@
...
 import akka.actor.typed.javadsl.Behaviors;
+// #test-header
 import akka.actor.testkit.typed.javadsl.ActorTestKit;
+
+// #test-header
 import akka.actor.testkit.typed.javadsl.TestProbe;
 import org.junit.AfterClass;
 import org.junit.Test;
+// #test-header
 import org.scalatest.junit.JUnitSuite;
```
[the remark this answers is not in the document]
Comment (helena) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: It seems to do

```java

import akka.actor.Scheduler;
import akka.actor.testkit.typed.javadsl.ActorTestKit;
import akka.actor.testkit.typed.javadsl.TestProbe;
import akka.actor.typed.ActorRef;
import akka.actor.typed.Behavior;
import org.junit.AfterClass;
import org.junit.Test;
import org.scalatest.junit.JUnitSuite;
import scala.util.Success;
import scala.util.Try;

import java.time.Duration;
import java.util.Objects;
import java.util.concurrent.CompletionStage;
import java.util.stream.IntStream;
```

File: akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java [hunk 2]
```diff
@@ -9,11 +9,17 @@
 import akka.actor.typed.Behavior;
 import akka.actor.typed.javadsl.AskPattern;
 import akka.actor.typed.javadsl.Behaviors;
+// #test-header
 import akka.actor.testkit.typed.javadsl.ActorTestKit;
+
+// #test-header
```
[the remark this answers is not in the document]
Comment (helena) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: It seems to do

```java
import akka.actor.Scheduler;
import akka.actor.testkit.typed.javadsl.ActorTestKit;
import akka.actor.testkit.typed.javadsl.TestProbe;
import akka.actor.typed.ActorRef;
import akka.actor.typed.Behavior;
import org.junit.AfterClass;
import org.junit.Test;
import org.scalatest.junit.JUnitSuite;
import scala.util.Success;
import scala.util.Try;

import java.time.Duration;
import java.util.Objects;
import java.util.concurrent.CompletionStage;
import java.util.stream.IntStream;

import static org.junit.Assert.assertEquals;
```

File: akka-actor-testkit-typed/src/test/scala/docs/akka/actor/testkit/typed/scaladsl/AsyncTestingExampleSpec.scala [hunk 3]
```diff
@@ -5,14 +5,20 @@
...
 import akka.actor.typed._
 import akka.actor.typed.scaladsl.AskPattern._
 import akka.actor.typed.scaladsl.Behaviors
 import akka.util.Timeout
+//#test-header
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.Matchers
 import org.scalatest.WordSpec
+
+//#test-header
```
[the remark this answers is not in the document]
Comment (helena) [reply akka-actor-testkit-typed/src/test/scala/docs/akka/actor/testkit/typed/scaladsl/AsyncTestingExampleSpec.scala]: It modifies slightly, also removing `akka.actor.typed.scaladsl.Behaviors`

```scala

import akka.actor.Scheduler
//#test-header
import akka.actor.testkit.typed.scaladsl.ActorTestKit

//#test-header
import akka.actor.typed._
import akka.actor.typed.scaladsl.AskPattern._
import akka.util.Timeout
//#test-header
import org.scalatest.BeforeAndAfterAll
import org.scalatest.Matchers
import org.scalatest.WordSpec

//#test-header
import scala.concurrent.Future
import scala.concurrent.duration._
import scala.util.Success
import scala.util.Try
```

File: akka-docs/src/main/paradox/typed/testing.md [hunk 4]
```diff
@@ -11,6 +11,19 @@ To use Akka TestKit Typed, add the module to your project:
...
+  artifact=scalatest_$scala.binary_version$
+  version=$scalatest.version$
+  scope=test
+}
+
+@@@
+
 ## Introduction
 
 Testing can either be done asynchronously using a real `ActorSystem` or synchronously on the testing thread using the `BehaviourTestKit`.
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/testing.md]: Needs to change to `BehaviorTestKit`:
```suggestion
Testing can either be done asynchronously using a real `ActorSystem` or synchronously on the testing thread using the `BehaviorTestKit`.
```

File: akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (helena) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: IntelliJ optimize imports does:
```java
import akka.actor.Scheduler;
import akka.actor.testkit.typed.javadsl.ActorTestKit;
import akka.actor.testkit.typed.javadsl.TestProbe;
import akka.actor.typed.ActorRef;
import akka.actor.typed.Behavior;
import org.junit.AfterClass;
import org.junit.Test;
import org.scalatest.junit.JUnitSuite;
import scala.util.Success;
import scala.util.Try;

import java.time.Duration;
import java.util.Objects;
import java.util.concurrent.CompletionStage;
import java.util.stream.IntStream;

import static org.junit.Assert.assertEquals;

// #test-header
// #test-header
// #test-header
// #test-header

// #test-header
```

File: akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (helena) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: It's deprecated anyway, and 'JUnitSuite has been moved from org.scalatest.junit to org.scalatestplus.junit`. @raboof I recall you ran into this a while back.

[the remark this answers is not in the document]
Comment (raboof) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: Yeah, scalatest is being replaced with scalatestplus, but a deprecated forwarder is still there. Unfortunately that forwarder doesn't work with 2.13.0-RC1... Indeed doesn't need to be visible for Java users though, will hide it.

File: akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (raboof) [reply akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java]: yeah I don't think IntelliJ optimize imports makes sense for files that are used for documentation snippets..

File: akka-docs/src/main/paradox/typed/testing.md (hunk 4 above)
Comment (raboof) [reply akka-docs/src/main/paradox/typed/testing.md]: Good catch! I'm replacing those with `@apidoc[BehaviorTestKit]` so they're checked at HTML generation time and link to the scala/javadoc

**Review by patriknw — APPROVED** [review]
LGTM

**Review by helena — APPROVED** [review]
LGTM

**Pull request merged**